rasa/core/domain.py

```diff
@@ -265,30 +265,61 @@ def collect_slots(slot_dict: Dict[Text, Any]) -> List[Slot]:
 
     @staticmethod
     def collect_intent_properties(
-        intents: List[Union[Text, Dict[Text, Any]]]
+        intents: List[Union[Text, Dict[Text, Any]]], entities: List[Text]
     ) -> Dict[Text, Dict[Text, Union[bool, List]]]:
         intent_properties = {}
         for intent in intents:
-            if isinstance(intent, dict):
-                name = list(intent.keys())[0]
-                for properties in intent.values():
-                    properties.setdefault("use_entities", True)
-                    properties.setdefault("ignore_entities", [])
-                    if (
-                        properties["use_entities"] is None
-                        or properties["use_entities"] is False
-                    ):
-                        properties["use_entities"] = []
-            else:
-                name = intent
+            if not isinstance(intent, dict):
                 intent = {intent: {"use_entities": True, "ignore_entities": []}}
```

Comment:
Not sure about this.

`intent` should be a dict with entries `use_entities` and `ignore_entities`. Before this line, `intent` either already is such a dict or is just a string, so this line simply "fixes" the intent variable if it was a string.

A reasonable alternative I could think of would be
`for incoming_intent in intents:
    if not isinstance(incoming_intent, dict):
        intent = {incoming_intent: {"use_entities": True, "ignore_entities": []}}
    else:
        intent = incoming_intent`

I would prefer to leave it as it is because I find it easier to read. It's a bit of a trade-off between simplicity and following conventions, I think. Do you think the alternative is better?

Reply:
Up to you, but I think your suggestion looks great 👍